heareval/embeddings/task_embeddings.py

```diff
@@ -0,0 +1,279 @@
+        return audio, self.filenames[idx]
+
+
+def get_dataloader_for_embedding(
+    data_path: Path, audio_dir: Path, embedding: Embedding, batch_size: int = 64
+):
+    if embedding.type == TORCH:
+        return DataLoader(
+            AudioFileDataset(data_path, audio_dir, embedding.sample_rate),
+            batch_size=batch_size,
+            shuffle=True,
+        )
+
+    elif embedding.type == TENSORFLOW:
+        raise NotImplementedError
+
+    else:
+        raise AssertionError("Unknown embedding type")
+
+
+def save_scene_embedding_and_label(
+    embeddings: np.ndarray, labels: Any, filename: Tuple, outdir: Path
+):
+    for i, file in enumerate(filename):
+        out_file = outdir.joinpath(f"{file}")
+        np.save(f"{out_file}.embedding.npy", embeddings[i])
+        np.save(f"{out_file}.label.npy", labels[i]["label"])
+
+
+def save_timestamp_embedding_and_label(
+    embeddings: np.ndarray,
+    timestamps: np.ndarray,
+    labels: np.ndarray,
+    filename: Tuple,
+    outdir: Path,
+):
+    for i, file in enumerate(filename):
+        out_file = outdir.joinpath(f"{file}")
+        np.save(f"{out_file}.embedding.npy", embeddings[i])
+        np.save(f"{out_file}.timestamps.npy", timestamps[i])
+        np.save(f"{out_file}.label.npy", labels[i])
+
+
+def get_labels_for_timestamps(
+    labels: List, timestamps: np.ndarray, vocab: pd.DataFrame
+) -> np.ndarray:
+    # Create a dictionary of the integer id for the label keyed on the str label.
+    # We want to do this so we can map from the str label to an integer and create
+    # a binary vector of labels for each timestamp.
+    # FIXME: the idx and label columns are switched. Needs to be fixed in the data
+    #   preprocessing pipeline and then updated here.
+    vocab = vocab.set_index("idx")
+    vocab = vocab.to_dict()["label"]
+
+    # Binary vector of labels for each timestamp
+    timestamp_labels = np.zeros((timestamps.shape[0], timestamps.shape[1], len(vocab)))
+
+    # TODO: make sure we are saving the labels in the correct units.
+    #   dcase gives units in seconds whereas the timestamps from the API
+    #   are in millseconds. Make sure dataset events are specified in ms.
+    assert len(labels) == len(timestamps)
+    for i, label in enumerate(labels):
+        tree = IntervalTree()
+        # Add all events to the label tree
+        for event in label:
+            # FIXME: Remove conversion to ms
+            tree.addi(1000 * event["start"], 1000 * event["end"], vocab[event["label"]])
+
+        # Update the binary vector of labels with intervals for each timestamp
+        for j, t in enumerate(timestamps[i]):
+            for interval in tree[t]:
+                timestamp_labels[i, j, interval.data] = 1.0
+
+    return timestamp_labels
+
+
+def task_embeddings(embedding: Embedding, task_path: Path):
+
+    metadata = json.load(task_path.joinpath("task_metadata.json").open())
+
+    # FIXME: idx and label columns are switched in labelvocab
```

Comment:
Let's use headers

Reply:
They are headers -- they are incorrect, needs to be fixed in the pipeline :)